app/uk/gov/hmrc/gform/core/parsers/BasicParsers.scala

```diff
@@ -99,6 +102,8 @@ object BasicParsers {
 
   lazy val positiveInteger: Parser[Int] = intParser("""\d+""")
 
+  lazy val nonZeroPositiveInteger: Parser[Int] = intParser("""^\d*[1-9]\d*$""")
```

Comment:
could this be simplified - `"""[1-9]{1}\d*"""`

Reply:
Done.

Reply:
Actually realized, this would not allow 01 and also match on non-numbers. I think your solution is the best one. Can you please revert?